<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mupdog</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dcgerard/mupdog">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<ul>
<li><a href="check_0_1.html">I think sufficient conditions are that R is psd and has diagonal entries less than or equal to 1.</a></li>
<li><a href="explore_mbb.html">Exploring a new generalization of the BB to allow for correlation.</a> The observations appear to be marginally beta-binomial, which is really cool. The sample correlation looks about the same as the “correlation” matrix used in the conditional distributions.</li>
<li><a href="check_bb_integral.html">The integral over the log BB density checks out</a></li>
<li><a href="try_mupdog_1.html">mupdog looks pretty good for a first try!</a> Interestingly, those inbreeding coefficients just don’t matter.</li>
<li><a href="try_mupdog_1000.html">mupdog still looks good when using 1000 Shirasawa SNPs</a>. No major difference here than with using 500 SNPs.</li>
<li><a href="no_cor.html">When contrain correlations to be 0, inbreeding still estimated to be 0.</a> Also, in high-bias SNP’s, some of the genotyping seems a little off.</li>
<li><a href="compare_cor_nocor.html">Benefits of fitting correlation mostly apparent in high-bias SNPs</a></li>
<li><a href="uitdewilligen.html">awesome results using the data from uidewilligen et al.</a> (1) The correlation matrix clearly clusters into two distinct groups. I wonder if there was a family in one group and the general population in the other. (2) High coverage SNPs look bad but moderate coverage SNP’s look fantastic. Are the high-coverage SNP’s due to something like copy number variation? (3) Inbreeding is more important in this non-sib population.</li>
<li><a href="depth.html">Looked at some filtering trends.</a> It seems that there is a higher incidence of bad-SNPs at the higher covered SNPs. Some generous filters seem to keep mostly good-looking SNPs.</li>
<li><a href="weird_ind.html">Individual 51 of the Uitdewilligen dataset has a super high od parameter on the genotypes.</a> I think its sample is corrupted. The readcounts always either all ref or all alt, even when all of the other individuals are all the other allele.</li>
<li><a href="uit_corr.html">Cannot get empirical estimated correlation to match estimated correlation.</a></li>
<li><a href="uitdewilligen_nocorr.html">When fitting without corr, get very different bias estimates and the genotyping results don’t look good where they differ.</a></li>
<li><a href="uitdewilligen_k2.html">Assuming diploid when tetraploid to uitdewilligen data results in lower objective value</a></li>
<li><a href="uitdewilligen_k6.html">Assuming hexaploid when tetraploid to uitdewilligen data results in lower objective value</a></li>
<li><a href="shir_fix_corr.html">Adding correlation helps in mislabelings in Shirasawa data</a></li>
<li><a href="corr_mle.html">I am updating the correlation matrix incorrectly.</a> I don’t know yet how much this matters.</li>
<li><a href="normal_bb.html">A non-1 variance biases the distribution away from <span class="math inline">\(K\mu\)</span>. This would make the allele frequencies harder to interpret.</a></li>
<li><a href="shirasawa_nodiag_constraint.html">Fit Shirasawa data without the covariance matrix constrained to be diagonal.</a> Can pick out the parent easily.</li>
<li><a href="shir_corr.html">You can understand the negative correlations by the fact that we are first removing the allele frequencies at each locus before calculating this correlation</a></li>
<li><a href="oracle_miss.html">Oracle misclassification error rates</a></li>
<li><a href="bias_or_remove.html">Oracle rates indicate that it’s better to account for bias directly rather than use WASP (which decreases the sample size).</a></li>
<li><a href="why_does_updog_do_poorly.html">Explore why updog does poorly on subset of data</a></li>
</ul>
<div id="looking-at-the-mupdog-distribution" class="section level1">
<h1>Looking at the <code>mupdog</code> distribution</h1>
<ul>
<li><a href="kl_div.html">The Gaussian Binomial can mimic the distribution of a convolution of two hypergeometrics in most cases, but it doesn’t do as well when the same variance needs to be shared across loci</a></li>
</ul>
</div>
<div id="flexdog" class="section level1">
<h1>Flexdog</h1>
<ul>
<li><a href="flexdog_try1.html"><code>mupdog</code> works as well or better than <code>flexdog</code> in every SNP I try in the Shirasawa dataset</a>. Note that <code>mupdog</code> is technically a mispecified model whereas <code>flexdog</code> is technically correctly specified (because the genotype distribution is unimodal). The problem seems to be that the bias parameter is estimated unstably. We need a better way to estimate it. Some sort of adaptive regularization?</li>
<li><a href="flexdog_v_updog.html"><code>flexdog</code> with option <code>model = &quot;hw&quot;</code> is same as <code>updog</code> with option <code>model = &quot;hw&quot;</code></a>.</li>
<li><a href="flex_v_up_speed.html"><code>flexdog</code> is slightly faster than <code>updog</code></a></li>
<li><a href="compare_ash_cvxr.html">Simple EM is faster than CVXR in unimodal update</a></li>
<li><a href="flexdog_try2.html">Compare CVXR and EM implementations of flexdog</a></li>
<li><a href="flexdog_v_updog_f1.html"><code>flexdog</code> with <code>model = &quot;s1&quot;</code> gives same results as <code>updog</code> with <code>model = &quot;s1&quot;</code> but is 3-6 times faster.</a></li>
<li><a href="ashpen.html">unimodal option relatively robust to penalty as long as it is large enough</a></li>
<li><a href="try_bb_option_flexdog.html">BB od is estimated near 0 when truth is S1</a>. This makes sense since F1 is even <em>less</em> dispersed than HWE.</li>
<li><a href="norm_flex.html">Normal option doesn’t work so well on SNP2</a></li>
<li><a href="oracle_fig.html">Number of samples needed to genotype with 95% accuracy</a></li>
<li><a href="out_compare.html"><code>flexdog</code> and <code>updog</code> match in terms of posterior summaries when using outliers</a>.</li>
<li><a href="check_for_ash_updog_err.html">Check for errors in the <code>&quot;ash&quot;</code> option in <code>updog</code></a>.</li>
<li><a href="oracle_tetra_joint.html">Joint dist with oracle for small read-depth tetraploid</a></li>
<li><a href="od_vs_rd.html">Relationship between read-depth and overdispersion estimates in shirasawa data</a>.</li>
<li><a href="play_uit.html">See how various priors behave on uitdewilligen data</a>. The flex prior works well sometimes and sometimes works horribly.</li>
<li><a href="uit_compare_variants.html">See how the calls made by uit differ from the calls made by <code>updog</code></a></li>
<li><a href="rocher_play.html">Play around with the Rocher et al data that compares 454 with GBS</a></li>
<li><a href="vcfdog_test.html">Proof of concept that <code>vcfdog</code> works</a></li>
<li><a href="cvs_vs_em.html"><code>updog</code> is faster than <code>ashr</code> in terms of weighted problem on small data</a></li>
<li><a href="cvx_vs_em2.html"><code>wem</code> is faster than <code>mixIP</code> and <code>mixEM</code></a></li>
<li><a href="hyper_is_mixture.html">Hypergeometric distribution is special case of mixtures</a></li>
<li><a href="explore_gatk.html">Some plots of the new simulation results which include GATK</a></li>
<li><a href="how_diff.html">Show how different preferential pairing can make the genotype distribution</a></li>
<li><a href="explore_s1pp.html">Some plots of the new s1pp simulation results</a></li>
<li><a href="shir_prop_mis.html">Histograms of estimated proportion of individuals misgenotyped in the shirasawa dataset</a></li>
</ul>
</div>
<div id="fitpoly" class="section level1">
<h1>fitPoly</h1>
<ul>
<li><a href="fit_poly_shir.html">Try <code>fitPoly</code> on the Shirasawa data</a>. It does just as well in genotyping, but the differences in precision estimates for each individual are very different.</li>
<li><a href="fitPoly_sim_play.html">In simulations, <code>updog</code> is better than <code>fitPoly</code> by many measures.</a></li>
<li><a href="fitPoly_on_shir.html">Example Shirasawa SNP where <code>updog</code> appears to perform much better than <code>fitPoly</code></a></li>
<li><a href="fitPoly_cor.html"><code>updog</code> always does better using correlation as the metric</a></li>
</ul>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
