<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="David Gerard" />

<meta name="date" content="2018-01-10" />

<title>Fit Shirasawa Dataset without diagonal constraint</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mupdog</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dcgerard/mupdog">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Fit Shirasawa Dataset without diagonal constraint</h1>
<h4 class="author"><em>David Gerard</em></h4>
<h4 class="date"><em>January 10, 2018</em></h4>

</div>


<div id="abstract" class="section level1">
<h1>Abstract</h1>
<p>I modified mupdog to not constrain the corrlation matrix to have a diagonal of ones. Here, I will test out how it performs on the Shirasawa SNPs.</p>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<p>Fit Shirasawa SNPs</p>
<pre class="r"><code>library(tidyverse)
library(mupdog)
library(corrplot)</code></pre>
<pre class="r"><code>refmat &lt;- as.matrix(read.csv(&quot;/home/david/Code/reproduce_genotyping/Output/shirasawa_snps/example_refcounts.csv&quot;, row.names = 1))
sizemat &lt;- as.matrix(read.csv(&quot;/home/david/Code/reproduce_genotyping/Output/shirasawa_snps/example_readcounts.csv&quot;, row.names = 1))
mout &lt;- mupdog::mupdog(refmat = refmat, sizemat = sizemat, ploidy = 6, verbose = TRUE, 
                       control = list(obj_tol = 10^-5), num_core = 7)
saveRDS(mout, file = &quot;../output/shir_fit/shir_fit_nodiag1.RDS&quot;)</code></pre>
<pre class="r"><code>mout &lt;- readRDS(file = &quot;../output/shir_fit/shir_fit_nodiag1.RDS&quot;)</code></pre>
<p>The diagonal of the estimated covariance matrix is always smaller than 1. This might be the model saying they are related.</p>
<pre class="r"><code>qplot(diag(mout$cor_mat), bins = 30, 
      main = &quot;histogram of diagonal estimates of covariance matrix&quot;) + 
  theme_bw()</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>There appears to be no specific groups.</p>
<pre class="r"><code>uR &lt;- cov2cor(mout$cor_mat)
corrplot(uR, method = &quot;color&quot;, order = &quot;hclust&quot;)</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>The histogram of correlations (using <code>cov2cor</code>) is almost exactly centered at 0 and spread between -4 and 4.</p>
<pre class="r"><code>qplot(uR[upper.tri(uR)], bins = 30) + theme_bw()</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>Plots look pretty good:</p>
<pre class="r"><code>set.seed(2)
for (index in 1:20) {
  pl &lt;- plot(mout, sample(1:1000, 1))
  print(pl)  
}</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-1.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-2.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-3.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-4.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-5.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-6.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-7.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-8.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-9.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-10.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-11.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-12.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-13.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-14.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-15.png" width="672" /></p>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-16.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-17.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-18.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-19.png" width="672" /><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-7-20.png" width="672" /></p>
<p>Let’s see if the allele frequencies make sense:</p>
<pre class="r"><code>qplot(mout$allele_freq, bins = 100) + theme_bw() +
  geom_vline(xintercept = 0:6/6, lty = 2)</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>This is really cool. It is saying that the allele frequency is either 1/6 or 5/6. Shirasawa et al did screening to make sure that all of their SNP’s were did not have ratios beyond AAAAAa or Aaaaaa. Apparently there aren’t any monoallelic SNPs in these data.</p>
<pre class="r"><code>emp_allele_freq &lt;- colMeans(mout$map_dosage) / 6

qplot(mout$allele_freq, emp_allele_freq) + theme_bw() +
  geom_vline(xintercept = 0:6/6, lty = 2) +
  geom_hline(yintercept = 0:6/6, lty = 2)</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Let’s look at the weird 0.75 allele-freq SNP (doesn’t look too bad).</p>
<pre class="r"><code>weird_snp &lt;- which.min((mout$allele_freq - 0.75)^2)
plot(mout, weird_snp)</code></pre>
<p><img src="shirasawa_nodiag_constraint_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>The parent is the individual that got the tiny variance estimate. Which makes super sense — in these data they always have allele AAAAAa or Aaaaaa!</p>
<pre class="r"><code>diag(mout$cor_mat)[1]</code></pre>
<pre><code>## [1] 0.0131</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
