<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="David Gerard" />

<meta name="date" content="2017-12-20" />

<title>uitdewilligen</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mupdog</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dcgerard/mupdog">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">uitdewilligen</h1>
<h4 class="author"><em>David Gerard</em></h4>
<h4 class="date"><em>December 20, 2017</em></h4>

</div>


<div id="abstract" class="section level1">
<h1>Abstract</h1>
<p>I try out mupdog on the data from</p>
<p>Uitdewilligen, Jan GAML, et al. “<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0062355">A next-generation sequencing method for genotyping-by-sequencing of highly heterozygous autotetraploid potato.</a>” PLoS One 8.5 (2013): e62355.</p>
<p>Which may be obtained here:</p>
<p><a href="https://doi.org/10.1371/journal.pone.0062355" class="uri">https://doi.org/10.1371/journal.pone.0062355</a></p>
</div>
<div id="extract-data-from-vcf" class="section level1">
<h1>Extract Data from VCF</h1>
<pre class="r"><code>library(vcfR)

uit &lt;- read.vcfR(file = &quot;~/Data/uitdewilligen_raw_data/NewPlusOldCalls.headed.vcf&quot;)

## DP = read-depth
## AA = alternative allele
## GT = genotype
## GQ = genotype quality
## GL = log-10 scaled likelihood of genotype
## RA = reference allele
## DP = read-depth
refmat  &lt;- extract.gt(uit, element = &quot;RA&quot;)
class(refmat) &lt;- &quot;numeric&quot;
altmat  &lt;- extract.gt(uit, element = &quot;AA&quot;)
class(altmat) &lt;- &quot;numeric&quot;
sizemat &lt;- extract.gt(uit, element = &quot;DP&quot;)
class(sizemat) &lt;- &quot;numeric&quot;

refmat[1000,10]
altmat[1000,10]
sizemat[1000, 10]


depth_vec &lt;- rowMeans(sizemat)
order_vec &lt;- order(depth_vec, decreasing = TRUE)[1:sum(depth_vec &gt;= 100, na.rm = TRUE)]

summary(depth_vec[order_vec])

saveRDS(file = &quot;../data/uitdewilligen_snps.RDS&quot;, list(refmat = t(refmat[order_vec, ]), sizemat = t(sizemat[order_vec, ]), ploidy = 4))</code></pre>
</div>
<div id="some-eda" class="section level1">
<h1>Some EDA</h1>
<pre class="r"><code>library(mupdog)
library(updog)
library(tidyverse)</code></pre>
<pre><code>## -- Attaching packages ------- tidyverse 1.2.1 --</code></pre>
<pre><code>## v ggplot2 2.2.1     v purrr   0.2.4
## v tibble  1.3.4     v dplyr   0.7.4
## v tidyr   0.7.2     v stringr 1.2.0
## v readr   1.1.1     v forcats 0.2.0</code></pre>
<pre><code>## -- Conflicts ---------- tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>inlist &lt;- readRDS(&quot;../data/uitdewilligen_snps.RDS&quot;)
refmat  &lt;- inlist$refmat
sizemat &lt;- inlist$sizemat
ploidy  &lt;- inlist$ploidy</code></pre>
<pre class="r"><code>set.seed(50)
for (index in sample(1:nrow(refmat), 20)) {
  pl &lt;- plot_geno(ocounts = refmat[, index], osize = sizemat[, index], ploidy = ploidy)
  print(pl)
}</code></pre>
<pre class="r"><code>mout &lt;- mupdog(refmat = refmat, sizemat = sizemat, ploidy = ploidy, verbose = TRUE, num_clust = 7,
               control = list(obj_tol = 10^-5, itermax = 200))
saveRDS(mout, &quot;../output/uit_fit/fit_uit1.RDS&quot;)</code></pre>
<pre class="r"><code>mout &lt;- readRDS(&quot;../output/uit_fit/fit_uit1.RDS&quot;)</code></pre>
<p>In non-sib populations, it seems that overdispersion on the genotypes is more important:</p>
<pre class="r"><code>qplot(mout$inbreeding, bins = 30) +
  theme_bw()</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Our prior for the sequencing error rate is a little off. Seems that it should have a mean of around -6 and a variance of around 0.5 (using mad to estimate this due to bad SNP’s).</p>
<pre class="r"><code>logitseq &lt;- log(mout$seq / (1 - mout$seq))
qplot(logitseq, bins = 30) +
  theme_bw()</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>mad(logitseq) ^ 2</code></pre>
<pre><code>## [1] 0.5232</code></pre>
<pre class="r"><code>median(logitseq)</code></pre>
<pre><code>## [1] -6.142</code></pre>
<p>The log-bias seems to have some heavy tails, but it is generally between -2 and 2, as our prior states. There seems to be a slight negative bias (in direction of reference), which makes sense.</p>
<pre class="r"><code>logbias &lt;- log(mout$bias)
qplot(logbias, bins = 30) +
  theme_bw()</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>I would remove all SNP’s with an overdispersion parameter greater than 0.05.</p>
<pre class="r"><code>qplot(mout$od, bins = 30) +
  theme_bw()</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>I don’t know why there is some negative correlation here.</p>
<pre class="r"><code>hist(mout$cor_mat[upper.tri(mout$cor_mat)])</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>There seems to be two distinct groups, which is super cool! Maybe there was a cluster of relatives here.</p>
<pre class="r"><code>library(corrplot)</code></pre>
<pre><code>## corrplot 0.84 loaded</code></pre>
<pre class="r"><code>corrplot(corr = mout$cor_mat, method = &quot;square&quot;, order = &quot;hclust&quot;)</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>The genes with the largest read depth look horrible</p>
<pre class="r"><code>for (index in 1:5) {
  pl &lt;- plot(mout, index)
  print(pl)
}</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-13-1.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-13-2.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-13-3.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-13-4.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-13-5.png" width="672" /></p>
<p>My guess is that these aren’t real genes, but a combination of multiple genes that look the same. E.g. this could be due to copy number variation.</p>
<p>The genes with moderate to small read depth look really good</p>
<pre class="r"><code>for (index in 20001:20005) {
  pl &lt;- plot(mout, index)
  print(pl)
}</code></pre>
<p><img src="uitdewilligen_files/figure-html/unnamed-chunk-14-1.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-14-2.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-14-3.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-14-4.png" width="672" /><img src="uitdewilligen_files/figure-html/unnamed-chunk-14-5.png" width="672" /></p>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
