<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="David Gerard" />

<meta name="date" content="2018-04-25" />

<title>Why Does Updog Do Poorly on Subsets of Data?</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mupdog</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dcgerard/mupdog">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Why Does Updog Do Poorly on Subsets of Data?</h1>
<h4 class="author"><em>David Gerard</em></h4>
<h4 class="date"><em>April 25, 2018</em></h4>

</div>


<div id="abtract" class="section level1">
<h1>Abtract</h1>
<p>Here, I explore why updog does poorly on a subset of the simulated data.</p>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## -- Attaching packages ------- tidyverse 1.2.1 --</code></pre>
<pre><code>## v ggplot2 2.2.1     v purrr   0.2.4
## v tibble  1.4.2     v dplyr   0.7.4
## v tidyr   0.7.2     v stringr 1.3.0
## v readr   1.1.1     v forcats 0.2.0</code></pre>
<pre><code>## -- Conflicts ---------- tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(updog)
simsdat &lt;- read.csv(&quot;../../../reproduce_genotyping/Output/sims_out/sims_out.csv&quot;)
badindex &lt;- which(simsdat$bias_val == 1 &amp; simsdat$od_param == 0.01 &amp; simsdat$uham &lt; 0.5) ## this is also the seed of one_rep</code></pre>
<p>Same way I calculated parvals in simulations</p>
<pre class="r"><code>## Read in size data to get realistic size distribution --------------------------------
size_mat   &lt;- read.csv( &quot;../../../reproduce_genotyping/Output/shirasawa_snps/example_readcounts.csv&quot;, row.names = 1)


## Parameters to explore --------------------------------------------
bias_seq  &lt;- c(1, 0.75, 0.5, 0.25)
seq_error &lt;- 0.005 ## Constant throughout
out_prop  &lt;- 0
od_seq    &lt;- c(0, 0.01, 0.05)
ploidy    &lt;- 6
itermax   &lt;- 1000

## Set up `updog` object parameters that don&#39;t vary ------------------
usame               &lt;- list()
usame$input$ploidy  &lt;- ploidy
usame$input$model   &lt;- &quot;hw&quot;
usame$seq_error     &lt;- seq_error
usame$out_prop      &lt;- out_prop
usame$p1geno        &lt;- -1
usame$p2geno        &lt;- -1
usame$out_mean      &lt;- 1/2
usame$out_disp      &lt;- 1/3
class(usame)        &lt;- &quot;updog&quot;


## Run Simulations ------------------------------------------------
parvals &lt;- expand.grid(allele_freq = seq(0.05, 0.95, length = itermax), bias_val = bias_seq, od_param = od_seq)
parvals$seed &lt;- 1:nrow(parvals)
parvals$osize &lt;- sapply(size_mat[, 1:itermax], function(x) x[!is.na(x)])

par_list &lt;- list()
for (list_index in 1:nrow(parvals)) {
  par_list[[list_index]] &lt;- list()
  for (inner_list_index in 1:(ncol(parvals) - 1)) {
    par_list[[list_index]][[inner_list_index]] &lt;- parvals[list_index, inner_list_index]
    names(par_list[[list_index]])[inner_list_index] &lt;- colnames(parvals)[inner_list_index]
  }
}

for (list_index in 1:nrow(parvals)) {
  par_list[[list_index]]$input$osize &lt;- parvals$osize[[list_index]]
}</code></pre>
<p>Run the one rep</p>
<pre class="r"><code>unew &lt;- par_list[[badindex]]


  set.seed(unew$seed)
  usim             &lt;- usame
  usim$bias_val    &lt;- unew$bias_val
  usim$allele_freq &lt;- unew$allele_freq
  usim$od_param    &lt;- unew$od_param
  usim$input$osize &lt;- unew$input$osize

  ## Simulate New Data ----------------------------------------------
  rout       &lt;- updog::rupdog(usim)
  ocounts    &lt;- rout$input$ocounts
  osize      &lt;- rout$input$osize
  true_ogeno &lt;- rout$ogeno

  ## Run updog -------------------------------------------------
  bias_start &lt;- exp(-2:2 * 0.7) ## plus to minus three sd
  llike_old &lt;- -Inf
  for (index in 1:length(bias_start)) {
    utemp &lt;- updog::updog_vanilla(ocounts = ocounts, osize = osize, ploidy = usim$input$ploidy, model = &quot;hw&quot;,
                                 out_prop = 0, update_outprop = FALSE, bias_val = bias_start[index], non_mono_max = Inf)
    if (utemp$llike &gt; llike_old) {
      uout &lt;- utemp
      llike_old &lt;- uout$llike
    }
  }</code></pre>
<p>Plot looks reasonable, just genotype 1 looks bad.</p>
<pre class="r"><code>plot(uout)</code></pre>
<p><img src="why_does_updog_do_poorly_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>uone &lt;- updog(ocounts = rout$input$ocounts, 
              osize = rout$input$osize, 
              ploidy = rout$input$ploidy, 
              model = &quot;f1&quot;, 
              bias_val = 1, 
              update_bias_val = FALSE)</code></pre>
<p>Fixing it at one has lower likelihood</p>
<pre class="r"><code>uone$llike</code></pre>
<pre><code>## [1] -744.2</code></pre>
<pre class="r"><code>uout$llike</code></pre>
<pre><code>## [1] -730.7</code></pre>
<p>But it does have nicer looking fits</p>
<pre class="r"><code>plot(uone)</code></pre>
<p><img src="why_does_updog_do_poorly_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>The bias is way off</p>
<pre class="r"><code>unew$bias_val</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>uout$bias_val</code></pre>
<pre><code>## [1] 1.597</code></pre>
<p>The 95% confidence interval for bias isnâ€™t too large</p>
<pre class="r"><code>uout$bias_val</code></pre>
<pre><code>## [1] 1.597</code></pre>
<pre class="r"><code>exp(uout$log_bias + 2 * sqrt(uout$covmat[1, 1]))</code></pre>
<pre><code>## [1] 1.799</code></pre>
<pre class="r"><code>exp(uout$log_bias - 2 * sqrt(uout$covmat[1, 1]))</code></pre>
<pre><code>## [1] 1.417</code></pre>
<p>od is way off</p>
<pre class="r"><code>unew$od_param</code></pre>
<pre><code>## [1] 0.01</code></pre>
<pre class="r"><code>uout$od_param</code></pre>
<pre><code>## [1] 0.02043</code></pre>
<p>But <code>prop_mis</code> will at least flag this SNP</p>
<pre class="r"><code>uout$prop_mis</code></pre>
<pre><code>## [1] 0.1916</code></pre>
<p>The genotypes are all shifted away from the truth</p>
<pre class="r"><code>ggplot(data_frame(geno = factor(rout$ogeno), pm = uout$postmean), aes(x = geno, y = pm)) +
  geom_boxplot() +
  theme_bw() +
  geom_abline(slope = 1, intercept = -1)</code></pre>
<p><img src="why_does_updog_do_poorly_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
