---
title: "Home"
output:
  html_document:
    toc: false
---

* [I think sufficient conditions are that R is psd and has diagonal entries less than or equal to 1.](check_0_1.html)
* [Exploring a new generalization of the BB to allow for correlation.](explore_mbb.html) The observations appear to be marginally beta-binomial, which is really cool. The sample correlation looks about the same as the "correlation" matrix used in the conditional distributions. 
* [The integral over the log BB density checks out](check_bb_integral.html)
* [mupdog looks pretty good for a first try!](try_mupdog_1.html) Interestingly, those inbreeding coefficients just don't matter.
* [mupdog still looks good when using 1000 Shirasawa SNPs](try_mupdog_1000.html). No major difference here than with using 500 SNPs.
* [When contrain correlations to be 0, inbreeding still estimated to be 0.](no_cor.html) Also, in high-bias SNP's, some of the genotyping seems a little off.
* [Benefits of fitting correlation mostly apparent in high-bias SNPs](compare_cor_nocor.html)
* [awesome results using the data from uidewilligen et al.](uitdewilligen.html) (1) The correlation matrix clearly clusters into two distinct groups. I wonder if there was a family in one group and the general population in the other. (2) High coverage SNPs look bad but moderate coverage SNP's look fantastic. Are the high-coverage SNP's due to something like copy number variation? (3) Inbreeding is more important in this non-sib population.
* [Looked at some filtering trends.](depth.html) It seems that there is a higher incidence of bad-SNPs at the higher covered SNPs. Some generous filters seem to keep mostly good-looking SNPs.
* [Individual 51 of the Uitdewilligen dataset has a super high od parameter on the genotypes.](weird_ind.html) I think its sample is corrupted. The readcounts always either all ref or all alt, even when all of the other individuals are all the other allele.
* [Cannot get empirical estimated correlation to match estimated correlation.](uit_corr.html)
* [When fitting without corr, get very different bias estimates and the genotyping results don't look good where they differ.](uitdewilligen_nocorr.html)
* [Assuming diploid when tetraploid to uitdewilligen data results in lower objective value](uitdewilligen_k2.html)
* [Assuming hexaploid when tetraploid to uitdewilligen data results in lower objective value](uitdewilligen_k6.html)
* [Adding correlation helps in mislabelings in Shirasawa data](shir_fix_corr.html)
* [I am updating the correlation matrix incorrectly.](corr_mle.html) I don't know yet how much this matters.
* [A non-1 variance biases the distribution away from $K\mu$. This would make the allele frequencies harder to interpret.](normal_bb.html)
* [Fit Shirasawa data without the covariance matrix constrained to be diagonal.](shirasawa_nodiag_constraint.html) Can pick out the parent easily.
* [You can understand the negative correlations by the fact that we are first removing the allele frequencies at each locus before calculating this correlation](shir_corr.html)
* [Oracle misclassification error rates](oracle_miss.html)
* [Oracle rates indicate that it's better to account for bias directly rather than use WASP (which decreases the sample size).](bias_or_remove.html)
